### ISSUE TITLE
header guard clang-tidy message is unactionable

### ISSUE THREAD

**Issue by mtrofin** [google/llvm-premerge-checks#205]
This message:

clang-tidy: warning: header guard does not follow preferred style [llvm-header-guard]

Is un-actionable: what is the preferred style? 

Suggestion: add a link to coding standard. FWIW, https://llvm.org/docs/CodingStandards.html has only to occurrences of "header guard", none illustrative.

**Comment by joker-eph** [comment]
FYI it documented here: https://github.com/llvm/llvm-project/blob/master/clang-tools-extra/docs/clang-tidy/checks/llvm-header-guard.rst

**Comment by mtrofin** [comment]
Ack - I hope that gets linked in the warning, for improved discoverability (which would make the message much more actionable, IMHO)

**Comment by metaflow** [comment]
@mtrofin I am not sure how pre-merge can improve this situation: tool is showing what clang-tidy is reporting. I believe it makes more sense to update clang-tidy warning and improve it for "normal" users as well. Or maybe I am missing something.

**Comment by mtrofin** [comment]
Ah, I see - so this is an issue for clang-tidy then.

**Comment by mtrofin** [comment]
Actually, thinking a bit more - I think there's a difference in what's reasonable to expect from the user if the user chooses to run the tool vs if the tool is run for the user, as the case is there. If I choose the run the tool, it's arguably fine to be expected to be familiar with it, and know/discover that, for instance, I can automatically fix violations. For example, I can run run-clang-tidy.py and pass "-fix". Basically, I don't need to know how to manually fix this violation, since it can be automatically fixed for me.

However, in this case, the tool is run for me and I am presented with a message. I'm not sure it's reasonable to expect the user is familiar with the clang-tidy tools' intricacies in this case. If, for instance, the message gave me the "Fix" or "Show Fix" prompt, that'd be ideal; alternatively, the UI could just show me how to repro and say "pass -fix to fix".

wdyt?

**Comment by metaflow** [comment]
Yes, I agree that showing fix is a better user experience. It might be a good idea for premerge checks to run clang tidy with fix and present possible solutions. As for "I run tool VS tool was run for me" maybe it would be nice to present some additional docs for new users, on the other hand I would rather not add any noise for experienced users.

**Comment by mtrofin** [comment]
What are the action items then - I suppose keeping this bug open here for the "run clang tidy with fix" part? The clang-tidy doc part is, I guess, a separate item. I could take a stab at that.
